WordPress/gutenberg
# Post Featured Image: Add "Use first image from post" option

## What?
This adds an option called "Use first image from post". When this option is enabled the Post Featured Image block will pull the first image from the post.

## Why?
For users with a lot of old posts without featured images this is a useful way to display images in the query block without having to manually add many featured images.

## How?
- Add a new attribute called `useFirstImageFromPost`.
- Add a toggle to the UI in the for the attribute.
- Add code to the block renderer to pull the first image from the post using the tag processor.
- Add code to the edit function to pull the first image from the post using `parse`.

## Testing Instructions
1. Create a new post with an image in it, but no Featured image.
2. Edit a template with a query block that contains a Featured image block.
3. Notice that the Query Block doesn't display any image in the Featured image block in the editor or the frontend.
4. Edit the Featured image block to enable the "Use first image from post".
5. Notice that the Query Block now displays the image in the Featured image block in the editor and the frontend.

## Screenshots or screencast 
<img width="286" alt="Screenshot 2023-11-27 at 16 19 09" src="https://github.com/WordPress/gutenberg/assets/275961/51102535-7dc6-4ee1-bc86-b6a8166c6569">
